RoslynPluginGenerator/NuGet/NuGetRepositoryFactory.cs

```diff
@@ -0,0 +1,68 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="NuGetRepositoryFactory.cs" company="SonarSource SA and Microsoft Corporation">
+//   Copyright (c) SonarSource SA and Microsoft Corporation.  All rights reserved.
+//   Licensed under the MIT License. See License.txt in the project root for license information.
+// </copyright>
+//-----------------------------------------------------------------------
+using NuGet;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace SonarQube.Plugins.Roslyn
+{
+    /// <summary>
+    /// Creates the remote repository from which the packages will be fetched
+    /// </summary>
+    public static class NuGetRepositoryFactory
+    {
+        /// <summary>
+        /// Returns the settings loaded from the user- and machine-wide config files
+        /// </summary>
+        public static ISettings GetSettingsFromConfigFiles()
```

Comment:
This method isn't easily testable: we can't easily control where _Settings.LoadDefaultSettings_ looks for config files (it looks in a per-user folder), and we don't really want our tests to create/delete the config files in those folders in case something goes wrong and leaves the machine in a bad state. I've manually tested this method.